Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1950,30 +1950,43 @@ void CGContextEOFillPath(CGContextRef context) {
 
 #pragma region Drawing Operations - CGImage
 
-/**
- @Status Interoperable
-*/
-void CGContextDrawImage(CGContextRef context, CGRect rect, CGImageRef image) {
-    NOISY_RETURN_IF_NULL(context);
-    NOISY_RETURN_IF_NULL(image);
+static HRESULT __CreateD2DBitmapFromCGImage(CGContextRef context, CGImageRef image, ID2D1Bitmap** bitmap) {
+    RETURN_HR_IF_NULL(E_INVALIDARG, context);
+    RETURN_HR_IF_NULL(E_INVALIDARG, image);
+    RETURN_HR_IF_NULL(E_POINTER, bitmap);
 
-    // Obtain the pixel format for the image
-    WICPixelFormatGUID imagePixelFormat = _CGImageGetWICPixelFormat(image);
+    ComPtr<IWICBitmap> bmap;
+    RETURN_IF_FAILED(_CGImageGetWICImageSource(image, &bmap));
 
-    CFRetain(image);
-    woc::unique_cf<CGImageRef> refImage;
-    refImage.reset(image);
+    ComPtr<ID2D1Bitmap> d2dBitmap;
```

Comment:
Oh, I see. I don't think this needs to be in two functions :P They'll always be found together.